src/Components/PostThought/PostThought.jsx

```diff
@@ -0,0 +1,67 @@
+import { useState } from "react";
+import { PostThoughtButton } from "../PostThoughtButton/PostThoughtButton";
+import { Validation } from "./Validation";
+import "./postThought.css";
+
+export const PostThought = ({ apiUrl, setThoughts, thoughts, handleThoughtFetch }) => {
+    // Creating a useState for the value of the textarea aka the thought meant to be posted
+    const [newMessage, setNewMessage] = useState("");
+    // Creating a useState for the errormessage that is to be shown in the p-tag of the validation component
+    const [errorMessage, setErrorMessage] = useState("");
+
+    // Function to change the state of thought/the value when an event occurs
+    const handleThoughtUpdate = (event) => {
+        setNewMessage(event.target.value);
+    }
+
+    const handleThoughtSubmit = async (event) => {
+        // Prevents default submit action of form
+        event.preventDefault();
+
+        try {
+            const response = await fetch(apiUrl, {
+                method: "POST",
+                headers: {
+                    "Content-Type": "application/json",
+                },
+                /* Makes the thought text to JSON format */
+                body: JSON.stringify({ message: `${newMessage}` }),
+            });
+
+            if (!response.ok) {
+                // Handle error, show an error-message in validation p-tag
+                throw new Error("Response was not ok");
+            }
+
+            const data = await response.json();
+            // Update the state with the new thought
+            setThoughts([data, ...thoughts]);
+            setNewMessage(""); // Clear the input field
+            setErrorMessage(""); // Clear any previous error messages
+            handleThoughtFetch();
+        } catch (error) {
+            console.error("Error posting thought", error);
+            // Set the error message state
+            setErrorMessage("Failed to post thought. Please try again.");
```

Comment:
I really like this part. I had some problem with the post posting anyway, but there was nothing in the post and when refreshing the page it disappeared. I will try this!